src/binary/ibc-setup/commands/channels.spec.ts

```diff
@@ -0,0 +1,278 @@
+const consoleLog = sinon.stub(console, 'log');
+
+const mnemonic =
+  'enlist hip relief stomach skate base shallow young switch frequent cry park';
+
+const registryYaml = `
+version: 1
+
+chains:
+  local_wasm:
+    chain_id: testing
+    prefix: wasm
+    gas_price: 0.025ucosm
+    rpc:
+      - http://localhost:26659
+  local_simapp:
+    chain_id: simd-testing
+    prefix: cosmos
+    gas_price: 0.025umuon
+    rpc:
+      - http://localhost:26658`;
+
+test.beforeEach(() => {
+  sinon.reset();
+});
+
+let channel: ChannelPair;
+let link: Link;
+
+test.before(async () => {
+  const ibcClientSimapp = await signingClient(simappChain, mnemonic);
+  const ibcClientWasm = await signingClient(wasmdChain, mnemonic);
+  link = await Link.createWithNewConnections(ibcClientWasm, ibcClientSimapp);
+  channel = await link.createChannel(
+    'A',
+    'transfer',
+    'custom',
+    Order.ORDER_UNORDERED,
+    'ics20-1'
+  );
+});
+
+test('lists channels for given chain (A)', async (t) => {
+  const logger = new TestLogger();
+
+  const options: Options = {
+    chain: 'local_wasm',
+    port: null,
+    connection: null,
+    mnemonic: null,
+    home: '/home/user',
+  };
+
+  fsReadFileSync.returns(registryYaml);
+
+  await run(options, (logger as unknown) as Logger);
+
+  const output = consoleLog.getCall(-1);
+
+  t.assert(
+    output.calledWithMatch(
+      new RegExp(
+        [
+          channel.src.channelId,
+          channel.src.portId,
+          link.endA.connectionID,
+          channelStateAsText(ChannelState.STATE_OPEN),
+        ].join('\\s+'),
+        'm'
+      )
+    )
+  );
+});
+
+test('lists channels for given chain (B)', async (t) => {
+  const logger = new TestLogger();
+
+  const options: Options = {
+    chain: 'local_simapp',
+    port: null,
+    connection: null,
+    mnemonic: null,
+    home: '/home/user',
+  };
+
+  fsReadFileSync.returns(registryYaml);
+
+  await run(options, (logger as unknown) as Logger);
+
+  const output = consoleLog.getCall(-1);
+
+  t.assert(
+    output.calledWithMatch(
+      new RegExp(
+        [
+          channel.dest.channelId,
+          channel.dest.portId,
+          link.endB.connectionID,
+          channelStateAsText(ChannelState.STATE_OPEN),
+        ].join('\\s+'),
+        'm'
+      )
+    )
+  );
+});
+
+test('filters channels by port', async (t) => {
+  const logger = new TestLogger();
+
+  const options: Options = {
+    chain: 'local_wasm',
+    port: channel.src.portId,
+    connection: null,
+    mnemonic: null,
+    home: '/home/user',
+  };
+
+  fsReadFileSync.returns(registryYaml);
+
+  await run(options, (logger as unknown) as Logger);
+
+  const output = consoleLog.getCall(-1).args[0] as string;
+
+  const channelWithDifferentPort = output
+    .split(os.EOL)
+    .slice(1) // remove table head
+    .find((value) =>
+      value.match(
+        new RegExp(
+          `[^\\s]+\\s+(?!${options.port})[^\\s]+\\s+[^\\s]+\\s+[^\\s]+`
```

Comment:
Hm, this is a little hard to understand and seems easy to make a mistake and get false negatives here. How about checking that every channel has the correct port?

Reply:
You're right. I inverted it and also spotted an edge case with last empty line.

I think it's easier to follow now, see: 281a400